packages/frontend/src/presentation/Navigator.tsx

```diff
@@ -1,34 +1,49 @@
-import {
-  createAppContainer,
-  createSwitchNavigator,
-  NavigationContainer,
-} from 'react-navigation';
+import React from 'react';
+import { NavigationNativeContainer } from '@react-navigation/native';
+import { createStackNavigator } from '@react-navigation/stack';
 import { Welcome } from './screens/Welcome';
 import { CardSearchFilter } from './screens/CardSearch/CardSearchFilter';
 import { CardSearchResults } from './screens/CardSearch/CardSearchResults';
 import { CardDetails } from './screens/CardSearch/CardDetails';
+import { Card } from '../domain/entities/Card';
 
 export const ROUTES = {
   WELCOME: 'Welcome',
-  HOME: 'Home',
   CARD_SEARCH_FILTER: 'CardSearchFilter',
   CARD_SEARCH_RESULTS: 'CardSearchResults',
   CARD_DETAILS: 'CardDetails',
 };
 
-export function createNavigator(): NavigationContainer {
-  return createAppContainer(
-    createSwitchNavigator(
-      {
-        Welcome,
-        CardSearchFilter,
-        CardSearchResults,
-        CardDetails,
-      },
-      {
-        initialRouteName: 'CardSearchFilter',
-        backBehavior: 'history',
-      },
-    ),
+export type RootParamList = {
+  Welcome: undefined;
+  CardSearchFilter: undefined;
+  CardSearchResults: { cardsFiltered: Card[] };
+  CardDetails: { card: Card };
+};
+
+const Stack = createStackNavigator<RootParamList>();
+
+export function Navigator(): React.ReactElement {
+  return (
+    <NavigationNativeContainer>
+      <Stack.Navigator initialRouteName="CardSearchFilter">
+        <Stack.Screen
+          options={{ headerShown: false }}
```

Comment:
Not a big problem, mas esse tipo de syntax cria um novo objeto toda vez que o o metodo render eh chamado. Assim sendo Stack.Screen sempre vai re renderizar porque options property eh diferente.